idaes/generic_models/properties/swco2.py

```diff
@@ -258,14 +247,14 @@ def rule_tc(b, p):
             )
             Ts = T/(251.196*pyunits.K)
             G = sum(b[i]/Ts**i for i in b)
-            return pyunits.convert(
-                (0.475598*pyunits.mW/pyunits.K**(3/2)/pyunits.m *
-                 sqrt(T)*(1 + 2.0/5.0*cint_over_k)/G) +
-                params.tc_d_1*rho[p] +
-                params.tc_d_2*rho[p]**2 +
-                params.tc_d_3*rho[p]**3 +
-                params.tc_d_4*rho[p]**4,
-                to_units=pyunits.W/pyunits.K/pyunits.m)
+            rho_unit = pyunits.kg/pyunits.m**3
+            return (
+                (0.475598/1e3 *
```

Comment:
Looking at this, the expression itself still has a mess of units spread throughout it (notably the cancelation of the density units). Based on that observation, I think there would be no performance loss by letting the parameters have the correct units (W/K.m) in the end.

The important change here is removing the `convert` statement and thus assuming that units are correct at this point (which they should be). This is probably where most of the performance hit was occurring, so making this assumption should solve the observed issue. However, because we made this assumption we would want to do a unit consistency check as part of our tests to confirm (but not as part of building the model in use). Due to that, I think that having units on the parameters would be a good idea, just to add a little more confirmation.

As an additional bonus, this would then allow users to use the `convert` methods to get the correct units if they ever chose to try to change the parameter values.

Reply:
Thanks @andrewlee94.  You're right about this.  I'll fix it.  I guess I just got in remove units mode and wasn't thinking clearly.